File: Data Structures & Algorithms/eating-bananas/submission-2.py
class Solution:
    def minEatingSpeed(self, piles: List[int], h: int) -> int:
        res = max(piles)
        l, r = 1, res # remember l = 1 not 0 because its rate actually
        while l <= r:
            mid = (l+r)//2
            sum = 0
            for p in piles:
                sum += math.ceil(float(p)/mid)
            if sum <= h:
                res = mid
                r = mid-1
            else:
                l = mid+1
        return res
        

        

        

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-2.java
class Solution {
    public int maxProfit(int[] prices) 
    {
        int i = 0;
        int j = i+1;
        int maxProfit = 0;

        while(j < prices.length)
        {
            maxProfit = Math.max(maxProfit, prices[j] - prices[i]);
            if(prices[j] < prices[i])
            {
                i=j;
            }
            j++;
        }

        return maxProfit;  
    }
}
